build.sc

```diff
@@ -87,72 +82,17 @@ object v {
     "-Xlint:infer-any",
     "-Xlint:missing-interpolator",
     "-language:reflectiveCalls",
-    s"-Wconf:${warnConf.mkString(",")}"
+    s"-Wconf:${scala2WarnConf.mkString(",")}"
   )
 }
 
-object utils extends Module {
-
-  val architecture = System.getProperty("os.arch")
-  val operationSystem = System.getProperty("os.name")
-
-  val mac = operationSystem.toLowerCase.startsWith("mac")
-  val linux = operationSystem.toLowerCase.startsWith("linux")
-  val windows = operationSystem.toLowerCase.startsWith("win")
-  val amd64 = architecture.matches("^(x8664|amd64|ia32e|em64t|x64|x86_64)$")
-  val aarch64 = architecture.equals("aarch64") | architecture.startsWith("armv8")
-
+private object utils extends Module {
   val firtoolVersion = {
     val j = _root_.upickle.default.read[Map[String, String]](os.read(millSourcePath / os.up / "etc" / "circt.json"))
     j("version").stripPrefix("firtool-")
   }
```

Comment:
how about moving this to `object v` since it defines the firtool version, which may be shared.